dotnet/src/SemanticKernel.Abstractions/AI/FunctionChoiceBehaviors/AutoFunctionChoiceBehavior.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Represent <see cref="FunctionChoiceBehavior"/> that provides either all of the <see cref="Kernel"/>'s plugins' function information to the model or a specified subset.
+/// This behavior allows the model to decide whether to call the functions and, if so, which ones to call.
+/// </summary>
+[Experimental("SKEXP0001")]
+public sealed class AutoFunctionChoiceBehavior : FunctionChoiceBehavior
+{
+    /// <summary>
+    /// List of the functions that the model can choose from.
+    /// </summary>
+    private readonly IEnumerable<KernelFunction>? _functions;
+
+    /// <summary>
+    /// Indicates whether the functions should be automatically invoked by the AI service/connector.
+    /// </summary>
+    private readonly bool _autoInvoke = true;
+
+    /// <summary>
+    /// This class type discriminator used for polymorphic deserialization of the type specified in JSON and YAML prompts.
+    /// </summary>
+    public const string TypeDiscriminator = "auto";
```

Comment:
One of the classes - [PromptExecutionSettingsTypeConverter](https://github.com/microsoft/semantic-kernel/blob/a3527d7aadb4ad75db437da8f6a3da3041a95ef4/dotnet/src/Functions/Functions.Yaml/PromptExecutionSettingsTypeConverter.cs#L31) that uses the constant lives in the `Functions.Yaml` project. But the `AutoFunctionChoiceBehavior` lives in `SemanticKernel.Abstractions` project. The const need to be public to allow the type converter to access it unless we want to go the `InternalsVisibleTo` way.

Alternatively, instead of getting the descriminator from the public const, it can be read from the `JsonPolymorphic` attribute the `FunctionChoiceBehavior` base class is annotated with using the System.Reflection.GetCustomAttribute method.

Reply:
removed